core/src/main/java/gyro/core/control/ForDirectiveProcessor.java

```diff
@@ -44,8 +46,8 @@ public void process(Scope scope, DirectiveNode node) {
             return;
         }
 
-        if (in instanceof List) {
-            List<?> list = (List<?>) in;
+        if (in instanceof List || in instanceof Set) {
```

Comment:
While I can't see a reason why a collection other than a `List` or `Set` would be passed in here, I wonder if we should make this `if (in instanceof Collection) {`?